### PR TITLE
Compositions expose

### PR DESCRIPTION
**Pull request by benjgil**
## Proposed Changes

- expose the Compositions component to enable displaying compositions anywhere in the UI (e.g. in a comparison tab with figma designs)
- add a customUI section to compositions, to enable passing some additional UI to be displayed below the composition - e.g. 
- customUI receives the componentModel and can then render according to this
- customUI can be prepended or post

### CODE REVIEW

File: scopes/compositions/compositions/compositions.tsx [hunk 1]
```diff
@@ -32,9 +32,13 @@ export type MenuBarWidget = {
   location: 'start' | 'end';
   content: ReactNode;
 };
-export type CompositionsProp = { menuBarWidgets?: CompositionsMenuSlot; emptyState?: EmptyStateSlot };
+export type CompositionsProp = {
+  menuBarWidgets?: CompositionsMenuSlot;
+  emptyState?: EmptyStateSlot,
+  customUI?: {prepend: boolean, component: (component: ComponentModel) => JSX.Element};
```
Comment (KutnerUri) [inline scopes/compositions/compositions/compositions.tsx:38]: no.
1. CustomUI === children
2. don't pass component as prop, they can use `ComponentContext` directly
3. prepend true/false is pain, just pass both options

how about this?
```
slots?: {
  menuItems?: ReactNode,
  preComposition?: ReactNode,
  postComposition?: ReactNode,
  end?: ReactNode // why do we need a slot after the split pane?
}
```

File: scopes/compositions/compositions/index.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ export { CompositionsAspect };
 export type { CompositionsMain } from './compositions.main.runtime';
 export type { CompositionsUI, CompositionsMenuSlot } from './compositions.ui.runtime';
 export { Composition, CompositionProps } from './composition';
+export { Compositions } from './compositions';
```
Comment (KutnerUri) [inline scopes/compositions/compositions/index.ts:7]: I'd extract this to a whole new component

**Review by KutnerUri — CHANGES_REQUESTED** [review]

File: scopes/compositions/compositions/compositions.tsx (hunk 1 above)
Comment (benjgil) [reply scopes/compositions/compositions/compositions.tsx:38]: 1. Not sure what you mean
2. good point
3. sounds good

Re the slots comment, you want to just merge all the slots together, so remove the separate parameter for menuItems?
If so sounds fine to me (though not sure what the question is in the comments about the end slot - think you added that...)

Comment (benjgil) [reply scopes/compositions/compositions/compositions.tsx:38]: Though i can actually see a use for a content slot after the split pane. Something that's generic for all the components

**Comment by benjgil** [comment]
Closing - implemented differently in the end
https://github.com/teambit/bit/pull/6074

**Pull request closed without merging**